src/cohesive_interpolations.jl

```diff
@@ -1,50 +1,30 @@
-Ferrite.nfacedofs(ip::SurfaceInterpolation) = Ferrite.ncelldofs(ip.ip_base) 
-Ferrite.ncelldofs(ip::SurfaceInterpolation) = 0 # cohesive elements never have dofs inside the cell
-
-# stuff for handling higher order elements correctly
-nvertices(::Interpolation{1,RefCube}) = 2
-nvertices(::Interpolation{2,RefCube}) = 4
-nvertices(::Interpolation{2,RefTetrahedron}) = 3
-nvertices(::Interpolation{3,RefCube}) = 8
-nvertices(::Interpolation{3,RefTetrahedron}) = 4
+Ferrite.getnbasefunctions(ip::SurfaceInterpolation{2, RefCube, order, Lagrange{1, RefCube, order}}) where {order} = 2*getnbasefunctions(ip.ip_base)
+Ferrite.vertexdof_indices(ip::SurfaceInterpolation{2, RefCube, order, Lagrange{1, RefCube, order}}) where {order} = ((1,),(2,),(order-1+3,),(order-1+4,)) # TODO we should be able to do it via calling `vertexdof_indices` of `ip_base` + arithmetic
+Ferrite.facedof_interior_indices(ip::SurfaceInterpolation{2, RefCube, order, Lagrange{1, RefCube, order}}) where {order} = (ntuple(i->2+i,order-1), ntuple(i->(order-1+4)+i,order-1)) # TODO we should be able to do it via calling `facedof_indices` of `ip_base` + arithmetic 
 
 struct JumpInterpolation{dim,shape,order,ip} <: SurfaceInterpolation{dim,shape,order,ip}
     ip_base::ip
 end
 
+nvertices(ip::Interpolation) = length(Ferrite.vertexdof_indices(ip))
+nvertexdofs(ip::Interpolation) = sum(length.(Ferrite.vertexdof_indices(ip)); init=0)
+nfacedofs(ip::Interpolation) = sum(length.(Ferrite.facedof_indices(ip)); init=0)
+ncelldofs(ip::Interpolation) = length(Ferrite.celldof_interior_indices(ip))
+
 JumpInterpolation(ip::Interpolation{dim,shape,order}) where {dim,shape,order} = JumpInterpolation{dim+1,shape,order,typeof(ip)}(ip)
-# no need to handle edge dofs, as underlying interpolation is <= 2D
+
 function Ferrite.value(ip::JumpInterpolation, i::Int, ξ::Vec{dim_base}) where dim_base
     (; ip_base) = ip
     @assert Ferrite.getdim(ip_base) == dim_base
     0 < i <= getnbasefunctions(ip) || throw(ArgumentError("no shape function $i for interpolation $ip"))
 
-    vertex_dofs = nvertices(ip_base)*Ferrite.nvertexdofs(ip_base)
-    face_dofs = length(Ferrite.faces(ip_base))*Ferrite.nfacedofs(ip_base)
-    cell_dofs = Ferrite.ncelldofs(ip_base)
-    if i <= 2vertex_dofs
-        if i <= vertex_dofs
-            return -Ferrite.value(ip_base, i, ξ)
-        else
-            return Ferrite.value(ip_base, i-vertex_dofs, ξ)
-        end
-    elseif i - 2vertex_dofs <= 2face_dofs 
-        if i-2vertex_dofs <= face_dofs 
-            return -Ferrite.value(ip_base, i-vertex_dofs, ξ)
-        else
-            return Ferrite.value(ip_base, i-vertex_dofs-face_dofs, ξ)
-        end
-    elseif i - 2vertex_dofs - 2face_dofs <= 2cell_dofs 
-        if i - 2vertex_dofs - 2face_dofs <= cell_dofs 
-            return -Ferrite.value(ip_base, i-vertex_dofs - face_dofs, ξ)
-        else
-            return Ferrite.value(ip_base, i-vertex_dofs-face_dofs-cell_dofs, ξ)
-        end
```

Comment:
What was the reasoning behind removing this block of logic? It was needed because the dofs are distributed 
1. all vertex dofs
2. all face dofs
etc.
That is still true after the recent changes to the interpolations / dof distribution, is it? This could be the cause why quadratic elements don't work any more.

Reply:
The initial version of the new dof logic was able to handle custom dof index orders (see above) and I have developed this PR against this version back then. This feature has been removed before merging because it interfered with high order dof distribution, which had higher priority for now. I will revert and check. Thanks!